app/concepts/Payment/wrapper/use_case/buy.rb

```diff
@@ -8,25 +8,25 @@ def initialize(wrapper)
           @wrapper = wrapper
         end
 
-        def call(id,password, ticket_params, seat, cinema_hall_id, movie_id)
-          left = @wrapper.getLeftRepository
-          right = @wrapper.getRightRepository
-          if validate?(id,password, left, right, seat,cinema_hall_id, movie_id)
-            success(left,ticket_params, cinema_hall_id, seat, movie_id)
+        def call(id,password, ticket_params)
+          left = @wrapper.left_Repository
```

Comment:
```suggestion
          left = wrapper.left_Repository
```



[attr_reader](https://mixandgo.com/learn/ruby_attr_accessor_attr_reader_attr_writer ) generates the [getter](https://www.codejava.net/coding/java-getter-and-setter-tutorial-from-basics-to-best-practices#What) for this attribute so you don't need access directly the instance variable.